shell/app/node_main.cc

```diff
@@ -110,17 +130,52 @@ int NodeMain(int argc, char* argv[]) {
   v8_crashpad_support::SetUp();
 #endif
 
-// TODO(deepak1556): Enable crashpad support on linux for
-// ELECTRON_RUN_AS_NODE processes.
-// Refs https://github.com/electron/electron/issues/36030
-#if BUILDFLAG(IS_WIN) || (BUILDFLAG(IS_MAC) && !IS_MAS_BUILD())
+#if BUILDFLAG(IS_LINUX)
+  auto os_env = base::Environment::Create();
+  std::string fd_string;
+  if (os_env->GetVar("CRASHDUMP_SIGNAL_FD", &fd_string)) {
+    base::StringToInt(fd_string, &crashdump_signal_fd);
+    ElectronCrashReporterClient::Create();
+    crash_reporter::InitializeCrashpad(false, "node");
+    crash_keys::SetCrashKeysFromCommandLine(
+        *base::CommandLine::ForCurrentProcess());
+    crash_keys::SetPlatformCrashKey();
+  }
+#elif BUILDFLAG(IS_WIN) || (BUILDFLAG(IS_MAC) && !IS_MAS_BUILD())
   ElectronCrashReporterClient::Create();
   crash_reporter::InitializeCrashpad(false, "node");
   crash_keys::SetCrashKeysFromCommandLine(
       *base::CommandLine::ForCurrentProcess());
   crash_keys::SetPlatformCrashKey();
 #endif
 
+#if BUILDFLAG(IS_LINUX)
+  // Shrink arguments without `--crashpad-handler-pid` to ensure that
+  // process.argv does not carry the flag.
+  int count = 0;
+  const std::string search_str =
+      std::string("--").append(crash_reporter::switches::kCrashpadHandlerPid);
+  for (int i = 0; i < argc; i++) {
+    if (base::StartsWith(argv[i], search_str)) {
+      auto results = base::SplitStringUsingSubstr(
+          argv[i], "=", base::TRIM_WHITESPACE, base::SPLIT_WANT_ALL);
+      base::StringToInt(results[1], &crashpad_handler_pid);
+      argv[i] = nullptr;
+      count += 1;
+    }
+  }
+
+  int new_argc = argc - count;
+  char** new_argv = new char*[new_argc];
+  for (int i = 0, j = 0; i < argc; i++) {
+    if (argv[i] != nullptr) {
+      new_argv[j++] = argv[i];
+    }
+  }
```

Comment:
oof, this seems a bit flimsy and error prone.

e.g. what if i spawn a process with something like

```js
child_process.fork('foo.js', ['--my-own', '--args="--foo --bar --crashpad-handler-pid=123"'])
```

this code will remove the --crashpad-handler-pid=xxx inside the --args switch's value, instead of removing the one at the end of the args. unlikely perhaps, but definitely incorrect.

Reply:
The above code is meant to remove all occurrences of the flag so it should handle the above case as well. The `base::StartsWith` should protect from the above example, it will not remove the flag inside `--args`.

Reply:
hm... i'd still prefer it if we could avoid this kind of string manipulation, it's very hard to get right.

Reply:
I think we can do the env variable passing and then adding it to `base::CommandLine` without polluting much of Nodejs internal like you suggested. Will make that change.

Reply:
Addressed in https://github.com/electron/electron/pull/36460/commits/21b0e1ffadfa7590a9deb814c2bf368c612cb336